app/routes/individual_response.py

```diff
@@ -0,0 +1,320 @@
+    name="individual_response", import_name=__name__, url_prefix="/individual-response/"
+)
+
+
+@individual_response_blueprint.before_request
+def before_individual_response_request():
+    metadata = get_metadata(current_user)
+    if not metadata:
+        raise NoTokenException(401)
+
+    logger.bind(
+        tx_id=metadata["tx_id"],
+        schema_name=metadata["schema_name"],
+        ce_id=metadata["collection_exercise_sid"],
+        questionnaire_id=metadata["questionnaire_id"],
+    )
+
+    logger.info(
+        "individual-response request", method=request.method, url_path=request.full_path
+    )
+
+    handle_language()
+
+    session_store = get_session_store()
+    g.schema = load_schema_from_session_data(session_store.session_data)
+
+
+@individual_response_blueprint.route("/", methods=["GET", "POST"])
+@login_required
+@with_questionnaire_store
+@with_schema
+def get_request_individual_response(schema, questionnaire_store):
+    router = Router(
+        schema,
+        questionnaire_store.answer_store,
+        questionnaire_store.list_store,
+        questionnaire_store.progress_store,
+        questionnaire_store.metadata,
+    )
+
+    if not router.can_access_hub():
+        return redirect("/questionnaire")
+
+    language_code = get_session_store().session_data.language_code
+    session_timeout = get_session_timeout_in_seconds(schema)
+
+    if request.method == "GET":
+        individual_id = request.args.get("individual_id")
+        previous_location_url = (
+            f"/questionnaire/household/{individual_id}/individual-interstitial/"
+        )
+
+        return render_template(
+            template="individual-response",
+            session_timeout=session_timeout,
+            language=language_code,
+            previous_location_url=previous_location_url,
+        )
+
+    individual_id = request.args.get("individual_id")
+
+    return redirect(
+        url_for(".get_individual_response_how", individual_id=individual_id)
+    )
+
+
+@individual_response_blueprint.route("/<individual_id>/how", methods=["GET", "POST"])
+@login_required
+@with_questionnaire_store
+@with_schema
+def get_individual_response_how(schema, questionnaire_store, individual_id):
+    router = Router(
+        schema,
+        questionnaire_store.answer_store,
+        questionnaire_store.list_store,
+        questionnaire_store.progress_store,
+        questionnaire_store.metadata,
+    )
+
+    if not router.can_access_hub():
+        return redirect("/questionnaire")
+
+    language_code = get_session_store().session_data.language_code
+    session_timeout = get_session_timeout_in_seconds(schema)
+    previous_location_url = f"/individual-response?individual_id={individual_id}"
+
+    first_name = questionnaire_store.answer_store.get_answer(
+        "first-name", individual_id
+    ).value
+    last_name = questionnaire_store.answer_store.get_answer(
+        "last-name", individual_id
+    ).value
```

Comment:
Given the individual section id is defined in the schema, should the answer id's also be defined there for consistency?. I feel like we lose in what ever we are trying to achieve by having a mix.

Reply:
What do you think @ajmaddaford & @LJBabbage? I agree with @MebinAbraham that these should go to the schema, but I want to get further opinions before moving these ids

Reply:
I'm not so sure. These would normally live in the block JSON for the page being rendered. Have you tried including them in that and using the existing code to render a block?

Reply:
This approach will become easier when I create the IR handler - shall we see what you think then @MebinAbraham ?

Reply:
spoken to @ajmaddaford about this - given we are now building the name using placeholder rendering defined in the block handler it's not necessary to put these into the schema